pkg/grizzly/resources.go

```diff
@@ -30,6 +30,13 @@ func (ref ResourceRef) String() string {
 // Resource represents a single Resource destined for a single endpoint
 type Resource struct {
 	Body map[string]interface{}
+
+	Provenance struct {
```

Comment:
Why using an anonymous type here? Having a `Provenance` type would also simplify the signature of `SetProvenance()`